src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,329 @@
+
+
+# def get_phonon_object(
+#     structure: Structure,
+#     supercell_matrix: np.array,
+#     displacement: float,
+#     sym_reduce: bool,
+#     symprec: float,
+#     use_standard_primitive: bool,
+#     kpath_scheme: str,
+#     code: str,
+# ):
+#     if code == "vasp":
+#         factor = VaspToTHz
+#     # TODO: add other codes?
+#
+#     cell = get_phonopy_structure(structure)
+#     if use_standard_primitive and kpath_scheme != "seekpath":
+#         phonon = Phonopy(
+#             cell,
+#             supercell_matrix,
+#             primitive_matrix=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+#             factor=factor,
+#             symprec=symprec,
+#             is_symmetry=sym_reduce,
+#         )
+#     else:
+#         phonon = Phonopy(
+#             cell,
+#             supercell_matrix,
+#             primitive_matrix="auto",
+#             factor=factor,
+#             symprec=symprec,
+#             is_symmetry=sym_reduce,
+#         )
+#     phonon.generate_displacements(distance=displacement)
+#     return phonon
+
+
+@job
+def generate_phonon_displacements(
+    structure: Structure,
+    supercell_matrix: np.array,
+    displacement: float,
+    sym_reduce: bool,
+    symprec: float,
+    use_standard_primitive: bool,
+    kpath_scheme: str,
+    code: str,
+):
+    """
+    Generate phonon displacements.
+
+    Parameters
+    ----------
+    phonopy_object: Phonopy_object
+
+    Returns
+    -------
+    List[Deformation]
+        A list of displacements.
+    """
+    cell = get_phonopy_structure(structure)
+    if code == "vasp":
+        factor = VaspToTHz
+    # a bit of code repetition here as I currently
+    # do not see how to pass the phonopy object?
+    if use_standard_primitive and kpath_scheme != "seekpath":
+        phonon = Phonopy(
+            cell,
+            supercell_matrix,
+            primitive_matrix=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+            factor=factor,
+            symprec=symprec,
+            is_symmetry=sym_reduce,
+        )
+    else:
+        phonon = Phonopy(
+            cell,
+            supercell_matrix,
+            primitive_matrix="auto",
+            factor=factor,
+            symprec=symprec,
+            is_symmetry=sym_reduce,
+        )
+    phonon.generate_displacements(distance=displacement)
+
+    supercells = phonon.supercells_with_displacements
+
+    displacements = []
+    for cell in supercells:
+        displacements.append(get_pmg_structure(cell))
+    return displacements
+
+
+@job(output_schema=PhononBSDOSDoc, data=[PhononDos, PhononBandStructureSymmLine])
+def generate_frequencies_eigenvectors(
+    structure: Structure,
+    supercell_matrix: np.array,
+    displacement: float,
+    sym_reduce: bool,
+    symprec: float,
+    use_standard_primitive: bool,
+    kpath_scheme: str,
+    code: str,
+    displacement_data: dict[str, list],
+    total_energy: float,
+    epsilon_static: Matrix3D = None,
```

Comment:
I wouldn't define all of these optional arguments here. Firstly, because it is duplication of default parameters and could result in the defaults getting out of sync with the document model. And secondly because if you want to add a new option to the document model you will have to remember to add it again here.

Instead just use  **kwargs and then change the function contents to something like:
```
    phonon_doc = PhononBSDOSDoc.from_forces_born(
         structure=structure,
         supercell_matrix=supercell_matrix,
         displacement=displacement,
         sym_reduce=sym_reduce,
         symprec=symprec,
         use_standard_primitive=use_standard_primitive,
         kpath_scheme=kpath_scheme,
         code=code,
         displacement_data=displacement_data,
         total_energy=total_energy,
         **kwargs
     )
```

Reply:
Thank you. I agree with this. I would however also add `born`, `epsilon_static` and `full_born` to the explicit keywords. They will be set in most of the cases.

Reply:
Ok that sounds good. Although please see my comment above about full_born, I think we should be able to detect this automatically.